plugins/modules/packaging/os/redhat_subscription.py

```diff
@@ -593,15 +593,22 @@ def update_subscriptions_by_pool_ids(self, pool_ids):
         consumed_pools = RhsmPools(self.module, consumed=True)
 
         existing_pools = {}
+        serials_to_remove = []
         for p in consumed_pools:
-            existing_pools[p.get_pool_id()] = p.QuantityUsed
+            pool_id = p.get_pool_id()
+            quantity_used = p.get_quantity_used()
+            existing_pools[pool_id] = quantity_used
+
+            quantity = pool_ids.get(pool_id, 0)
+            if quantity == 0 or quantity is not None and quantity != quantity_used:
+                serials_to_remove.append(p.Serial)
 
-        serials_to_remove = [p.Serial for p in consumed_pools if pool_ids.get(p.get_pool_id(), 0) != p.QuantityUsed]
         serials = self.unsubscribe(serials=serials_to_remove)
 
         missing_pools = {}
         for pool_id, quantity in sorted(pool_ids.items()):
-            if existing_pools.get(pool_id, 0) != quantity:
+            quantity_used = existing_pools.get(pool_id, 0)
+            if quantity_used == 0 or quantity is not None and quantity_used != quantity:
```

Comment:
Shouldn't this condition be
```suggestion
            if (quantity is None and quantity_used == 0) or (quantity is not None and quantity_used != quantity):
```
instead? (I added the brackets to make it easier to read, they are not necessary since `and` is higher priority than `or`.)

If `quantity_used == 0`, you only want to create a pool when `quantity != 0` - but that's already handled in the second part of the condition if `quantity is not None`, so the first part only needs to handle `quantity_used == 0 and quantity is None`.

Reply:
I hope, that I have not missed something, but I think both conditional statements behave identically. Have you come to another conclusion? In case of `quantity_used = 0`, it is not important, if the user has provided the required amount of entitlements or not, so why should we add it to the first part of the conditional statement and force the evaluation of the second conditional part? We already know that we must do something and can save yourselves the second evaluation.

What do you think?

Reply:
When `quantity == 0` and `quantity_used == 0`, your condition `quantity_used == 0 or quantity is not None and quantity_used != quantity` evaluates to `True`, while my condition `(quantity is None and quantity_used == 0) or (quantity is not None and quantity_used != quantity)` evaluates to `False`.

Reply:
Ah, yes. I have not considered this case. Under these circumstances, the system is currently not subscribed to the pool and should not be subscribed in the next run because of the user has provided `quantity = 0` 🤗.